### PR TITLE
TYP: fix mypy ignored error in pandas/io/formats/latex.py

### PR DESCRIPTION
**Pull request by ivanovmg**
- [ ] xref #37715 
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Handle mypy ignored error in ``pandas/io/formats/latex.py``.

### CODE REVIEW

File: pandas/io/formats/latex.py [hunk 1]
```diff
@@ -73,10 +73,8 @@ def __init__(
         self.multicolumn_format = multicolumn_format
         self.multirow = multirow
         self.clinebuf: List[List[int]] = []
-        self.strcols = self._get_strcols()
-        self.strrows: List[List[str]] = list(
-            zip(*self.strcols)  # type: ignore[arg-type]
-        )
+        self.strcols: Sequence[Sequence[str]] = self._get_strcols()
+        self.strrows: Sequence[Sequence[str]] = list(zip(*self.strcols))
```
Comment (simonjayhawkins) [inline pandas/io/formats/latex.py:77]: The variable type annotations can be removed. My preference is not to add variable types annotations unless absolutely necessary.

the variable types for self.strcols and self.strrows can (should) be inferred from the rhs expression.

self.strcols:  Revealed type is 'builtins.list[builtins.list[builtins.str]]'
self.strrows: Revealed type is 'builtins.list[builtins.tuple*[Any]]'

I'm not sure why mypy can't infer more precise for self.strrows

mypy is green, but if wanted to be more precise for self.strrows could use `List[Tuple[str, ...]]`

**Review by simonjayhawkins — APPROVED** [review]
Thanks @ivanovmg generally lgtm.

File: pandas/io/formats/latex.py (hunk 1 above)
Comment (ivanovmg) [reply pandas/io/formats/latex.py]: Thank you for the suggestions!
Updated accordingly.

**Review by simonjayhawkins — APPROVED** [review]
Thanks @ivanovmg ping on green.

**Comment by ivanovmg** [comment]
@simonjayhawkins looks like green (internal Windows py38_np18 error)

**Comment by simonjayhawkins** [comment]
Thanks @ivanovmg

**Pull request merged**